torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1253,27 +1278,35 @@ def _get_key(self):
         """
         return (len(self.replacements), len(self.divisible))
 
-    def _produce_dyn_sizes(self, ex: torch.Tensor, source: Source) -> List[sympy.Expr]:
+    def _produce_dyn_sizes(self,
+                           ex: torch.Tensor,
+                           source: Source,
+                           dims: Optional[List[DIM_DYNAMISM_STATE]]) -> List[sympy.Expr]:
         from torch._dynamo.source import TensorPropertySource, TensorProperty
         size = []
         for i, val in enumerate(ex.size()):
-            is_dynamic = _is_dim_dynamic(ex, i)
-            if _should_allocate(is_dynamic, self.assume_static_by_default):
+            dim_state = dims[i] if dims else DIM_DYNAMISM_STATE.DUCK
+            is_static = dim_state == DIM_DYNAMISM_STATE.STATIC
+            if not is_static:
                 size.append(self.create_symbol(
-                    val, TensorPropertySource(source, TensorProperty.SIZE, i), is_dynamic
+                    val, TensorPropertySource(source, TensorProperty.SIZE, i), dim_state
                 ))
             else:
                 size.append(sympy.Integer(val))
         return size
 
-    def create_symbolic_sizes_strides_storage_offset(self, ex: torch.Tensor, source: Source):
+    def create_symbolic_sizes_strides_storage_offset(self,
+                                                     ex: torch.Tensor,
+                                                     source: Source,
+                                                     dynamic_dims: Optional[List[DIM_DYNAMISM_STATE]],
```

Comment:
You pass these as kwargs in a bunch of other places, for consistency make these kwargs only too?

Reply:
sure thing